techmatters/flex-plugins
# CHI-2749: Fix request spam

## Description

Fix a number of small issues that contribute to getting the Flex plugin into a request loop

* Disable case summary and case save buttons when a request is in flight
* Fix the routing redux logic so routes cannot be completely emptied
* Fix logic issue in TaskView.tsx where it always tries to retrieve a contact if there is no route - this makes no sense because retrieving a contact won't create a route

### Checklist
- N/A Feature flags added
- N/A Strings are localized

### Other Related Issues

None

### Verification steps

1. Log into Aselo
2. Start creating an offline contact
3. Add a new case to the offline contact
4. Open Developer Tools and go to the 'Network' tab
5. Set Network Throttling to 'Slow 3G'
6. Click 'Edit on the Case Summary and make a change to the case summary text
7. Click 'Save' as many times as you can before the button disappears

Expected Behaviour:

The button should disable after the initial click, and only 1 PUT request to ../case/###/overview should be logged on the nework tab
Repeated requests to the GET 'byTaskSid/offline-task-WK-###' and GET cases/### should NOT be being loggged in the network tab

### AFTER YOU MERGE

1. Cut a release tag using the Github workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
2. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
3. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P
